File: Python/construct_the_rectangle.py
# https://leetcode.com/problems/construct-the-rectangle/

from math import sqrt

from utils.examination import Examiner, Testcase, Logger


class Solution:
    def constructRectangle(self, area: int) -> list[int]:
        square: int = int(sqrt(area))
        for i in range(square, 0, -1):
            if 0 == area % i:
                return [area // i, i]

def examine() -> None:
    Examiner(Solution(), 'constructRectangle').run((
        Testcase((4,), [2,2]),
        Testcase((37,), [37,1]),
        Testcase((122122,), [427,286]),
    ))

File: TypeScript/solutions/construct-the-rectangle.ts
// https://leetcode.com/problems/construct-the-rectangle/
import { Examiner, Testcase } from '../utils';


function constructRectangle(area: number): number[] {
    const square = Math.floor(Math.sqrt(area));
    for (let i = square; i >= 1; i--) {
        if (area % i === 0) return [area / i, i];
    }
    return [];
}

export function examine(): void {
    new Examiner(constructRectangle).run([
        new Testcase([4], [2, 2]),
        new Testcase([37], [37, 1]),
        new Testcase([122122], [427, 286]),
    ]);
}
